core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1066,6 +1066,31 @@ public void listStatus(AlluxioURI path, ListStatusContext context,
     return fileInfos;
   }
 
+  /**
+   * {@link DefaultFileSystemMaster} has two private methods associated with sync metadata:
+   * 1. {@link DefaultFileSystemMaster#loadMetadataIfNotExist} sync metadata in load-only mode,
+   * 2. {@link DefaultFileSystemMaster#syncMetadata} sync metadata not in force mode.
+   * Due to loadMetaData need work on both non-load-only mode and force mode, we need custom a
+   * InodeSyncStream object which foreSync param is true and loadOnly param is false.
+   * Further more, synchronize metadata may modify the structure of the inode tree, the lock
+   * pattern of the InodeSyncStream object should be LockPattern.WRITE_EDGE.
+   */
+  @Override
+  public void loadMetaData(AlluxioURI path, LoadMetadataContext context)
+      throws AccessControlException, FileDoesNotExistException, InvalidPathException, IOException {
+    try (RpcContext rpcContext = createRpcContext(context)) {
+      DescendantType syncDescendantType =
+          GrpcUtils.fromProto(context.getOptions().getLoadDescendantType());
+      FileSystemMasterCommonPOptions commonOptions = context.getOptions().getCommonOptions();
+      InodeSyncStream sync =
+          new InodeSyncStream(new LockingScheme(path, LockPattern.WRITE_EDGE, true), this,
```

Comment:
@gpang  to sync metadata, do we need to start with `LockPattern.WRITE_EDGE`? can we start with `READ`?
Also, can you check if the parameters for `InodeSyncStream` here is correct?

Reply:
I think it can be READ too. However, shouldn't this behave the same as `listStatus` until it is time to return results? I feel this command should be just the prefix of the code for `listStatus`. Alternatively, should there just be an option to `listStatus` which ignores sending results, or something like "forceLoadOnly"? There would be a lot of shared code between load metadata and list status.

Reply:
thanks. @JySongWithZhangCe  based on Gene's feedback, let's try to emulate what `listStatus` is doing and reuse code as much as possible. 

Meanwhile, @gpang , I agree with you that it can be a new option in `ListStatus` rpc. However, given the code we are adding a minimal, I view it is cleaner to have a separate RPC for its own purpose rather than complicating listStatus which is already pretty complicated. Let me know if you feel it strongly and we can discuss